services/ux-backend/handlers/devicefinder/handler.go

```diff
@@ -0,0 +1,174 @@
+package devicefinder
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/red-hat-storage/ocs-operator/v4/services/devicefinder"
+	"github.com/red-hat-storage/ocs-operator/v4/services/ux-backend/handlers"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// DiscoveryRequest represents the request body for device discovery
+type DiscoveryRequest struct {
+	Namespace    string               `json:"namespace"`
+	NodeSelector *corev1.NodeSelector `json:"nodeSelector,omitempty"`
+	Tolerations  []corev1.Toleration  `json:"tolerations,omitempty"`
+}
+
+// DiscoveryResponse represents the response for device discovery
+type DiscoveryResponse struct {
+	Message string                     `json:"message"`
+	Success bool                       `json:"success"`
+	Devices map[string]DiscoveryResult `json:"devices,omitempty"`
+}
+
+// DiscoveryResult represents the result of device discovery for a node
+type DiscoveryResult struct {
+	DiscoveredTimeStamp string             `json:"discoveredTimeStamp,omitempty"`
+	DiscoveredDevices   []DiscoveredDevice `json:"discoveredDevices"`
+}
+
+// DiscoveredDevice shows the list of discovered devices with their properties
+type DiscoveredDevice struct {
+	DeviceID string `json:"deviceID"`
+	Path     string `json:"path"`
+	Model    string `json:"model"`
+	Type     string `json:"type"`
+	Vendor   string `json:"vendor"`
+	Size     int64  `json:"size"`
+	WWN      string `json:"WWN"`
+}
+
+func HandleMessage(w http.ResponseWriter, r *http.Request, cl client.Client, namespace string) {
+	switch r.Method {
+	case "POST":
+		handlePost(w, r, cl, namespace)
+	case "GET":
+		handleGet(w, r, cl, namespace)
+	default:
+		handleUnsupportedMethod(w, r)
+	}
+}
+
+func handlePost(w http.ResponseWriter, r *http.Request, cl client.Client, defaultNamespace string) {
+	// Parse request body
+	var req DiscoveryRequest
+	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
+		klog.Errorf("failed to decode request body: %v", err)
+		http.Error(w, "Invalid request body", http.StatusBadRequest)
+		return
+	}
+
+	// Use namespace from request or default
+	targetNamespace := req.Namespace
+	if targetNamespace == "" {
+		targetNamespace = defaultNamespace
+	}
+
+	// Create device finder daemonset
+	err := devicefinder.CreateDeviceFinderDaemonSet(r.Context(), cl, targetNamespace, req.NodeSelector, req.Tolerations)
+	if err != nil {
+		klog.Errorf("failed to create device finder daemonset: %v", err)
+		response := DiscoveryResponse{
+			Message: fmt.Sprintf("Failed to create device finder daemonset: %v", err),
+			Success: false,
+		}
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusInternalServerError)
+		json.NewEncoder(w).Encode(response)
+		return
+	}
+
+	// Return success response
+	response := DiscoveryResponse{
+		Message: "Device finder daemonset created successfully",
+		Success: true,
+	}
```

Comment:
IMO we dont need to sent this in response, send 200OK from here, we dont need to show what went wrong to the user.